### DISCUSSION TITLE
CRSFforArduino RF mode or power in stats example?

### DISCUSSION THREAD

**Discussion by Dmack103** [ZZ-Cat/CRSFforArduino#144] [Q&A] [answered]
I am trying to figure out how to modify the Example code for Stats sketch to show transmit power.  Any ideas?  I took a stab in the dark and tried to print "rf_mode", but this didn't work.

**Comment by ZZ-Cat** [comment, marked as answer]
The [transmission power](https://github.com/ZZ-Cat/CRSFforArduino/blob/Main-Trunk/examples%2Flink_stats%2Flink_stats.ino#L82-L98) already exists in the link statistics telemetry.

There is no need to modify what is already there to begin with.

Also, `rf_mode` is not yet implemented, as that is a CRSF v3 thing. CFA currently implements CRSF v2, which does not have `rf_mode` telemetry.

**Comment by Dmack103** [comment]
Sorry, I was looking for Packet Rate or RF_Mode, not power.  My mistake.

Thanks for letting me know it is not available.  I didn't even know that
CRSF is being updated.

**Reply by ZZ-Cat** [reply]
Yea, nah. Neither did I until [this](https://github.com/tbs-fpv/freedomtx/issues/26#issuecomment-2479285161) came out in November. There had been a lot of requests for Team BlackSheep to create an official repository for the CRSF Protocol and a _lot_ of back-and-forth from other (more prominent) developers from other projects (EG Betaflight, RotorFlight, ExpressLRS etc) chiming in on that thread spurring TBS to get onto it. I am grateful they got it done, in the finish.

Originally, I was using [this](https://github.com/crsf-wg/crsf/wiki) as my point-of-reference to implement the protocol in its entirety. Moving forward, I will refer to [this](https://github.com/tbs-fpv/tbs-crsf-spec/blob/main/crsf.md) after I have released [Version 1.1.0 of CRSF for Arduino](https://github.com/ZZ-Cat/CRSFforArduino/issues/111).

Currently, I am on hiatus until early 2025.

**Comment by Dmack103** [comment]
Thanks for the info and enjoy the break.On Dec 10, 2024, at 2:36 PM, ZZ-Cat ***@***.***> wrote:﻿
Yea, nah. Neither did I until this came out in November. There had been a lot of requests for Team BlackSheep to create an official repository for the CRSF Protocol and a lot of back-and-forth from other (more prominent) developers from other projects (EG Betaflight, RotorFlight, ExpressLRS etc) chiming in on that thread spurring TBS to get onto it. I am grateful they got it done, in the finish.
Originally, I was using this as my point-of-reference to implement the protocol in its entirety. Moving forward, I will refer to this after I have released Version 1.1.0 of CRSF for Arduino.
Currently, I am on hiatus until early 2025.
